service-loadbalancer/README.md

```diff
@@ -15,6 +15,35 @@ This module will not help with the latter
 
 ## Overview
 
+The image below roughly describes the service-loadbalancer alternative (compared with using NodePorts).
+
+```
+                          +----------------------+                     
+                          | haproxy load balancer|                     
+                          +---------+------------+                     
+                                    |                                  
+                +-------------------+---------------+                  
+                |                                   |                  
+   +------------+----------------+     +-----------+--------------------+
+   |kubelet role=loadbalancer (1)|     |kubelet role=loadbalancer (2)   |
+   +----+------------------------+     +--------------------------------+
+        |                                                       |
+        |                                                       |
+        | --tcpServices=postgresql:3333                         | --tcpServices=postgresql:3333  
+        |                                                       |    
+        |               +-----------------------------+         |    
+        |               |   service for postgresql    |         |    
+        |               +-----------------------------|         |    
+        |                      |  targetPort=2222,port=3333     |    
+        |           +------+-----------+    +----------------+  |    
+        |           |                  |    |                |  |    
+        +---------->| k1postgresql     |    | k2postgresql   | <-  
+                    | container        |    | container      |
+                    | Port=2222        |    | Port=2222      |          
+                    +------------------+    +----------------+          
+Direct load balanced connections  to kubelets, without need for NodePort on every kubelet.
```

Comment:
This is a good point, by endpoint, you mean the containerPort, correct?  Since that container is running in a conceptual "pod", that is why its shown this way.  Maybe I'm missing something?